dist/profile/manifests/buildmaster.pp

```diff
@@ -306,6 +306,14 @@
 RewriteCond %{HTTP_USER_AGENT} YisouSpider|Catlight*|CheckmanJenkins [NC]
 RewriteRule ^.* \"https://jenkins.io/infra/ci-redirects/\"  [L]
 
+# Block requests which attempt to access API endpoints and end up hanging
+# Jenkins. Really Jenkins should have a permission to block Anonymous/API
+# access but it doesn't, so Apache it is.
+# See also: https://issues.jenkins-ci.org/browse/INFRA-1204
+RewriteCond %{REQUEST_FILENAME} ^(.*)api/json(.*)$ [NC]
```

Comment:
@orrc For completeness' sake, perhaps. None of the slow requests that I found were hitting api/xml though.


@olblak Since i'm going to be offline for a bit, would you mind adjusting the PR accordingly and getting it into production?